### PR TITLE
Use the official pages mirror and not the github repository for caching pages

### LINKED ISSUE #31: Caching should use resources from tldr.sh and not clone a git repositroy
The minimum specification for tldr clients (https://github.com/tldr-pages/tldr/wiki/Minimum-specifications-for-tldr-pages-command-line-clients) states that caching should not be done via github.com but via tldr.sh. 
I propose caching tldr.sh/assets/tldr.zip.

Advantages:
* No dependency on git
* less storage overhead
* git repositories were never meant for such purposes

I can look into implementing this in a few days.

### PR DESCRIPTION
**Pull request by gutjuri**
closes #31 

`git` is no longer a dependency. 

Performance is not affected. The initialisation of the cache is much faster now. Even though updating the cache is no longer incremental, it is hardly slower.

### CODE REVIEW

**Comment by psibi** [comment]
That was quick. Thanks! I'm liking this so far. I will review the MR once the CI passes.

**Comment by psibi** [comment]
I'm not seeing anywhere where the `--update` workflow is handled ? Is that intended ?

**Comment by gutjuri** [comment]
What exactly are you referring to? Starting the programme with `--update` invokes the code I have provided, if that's what you mean.

**Comment by psibi** [comment]
@gutjuri Ah, I see what you mean. Thanks! I tested your MR locally, and in it's initial use it outputs this:

```
$ tldr ls
No tldr entry for ls
```

Probably check whether the directory is initialized and if not try downloading and populating the cache ?

Also, there seems to be a speed difference of 3 second between the git version and the zip version of `tldr --update`. But that seems fine, I guess. Not having dependency on `git` is nice!

**Comment by gutjuri** [comment]
> @gutjuri Ah, I see what you mean. Thanks! I tested your MR locally, and in it's initial use it outputs this:
> 
> ```
> $ tldr ls
> No tldr entry for ls
> ```
> 
> Probably check whether the directory is initialized and if not try downloading and populating the cache ?
> 

Done. I also added an auto-update feature that updates the cache when it is older than 7 days.

> Also, there seems to be a speed difference of 3 second between the git version and the zip version of `tldr --update`. But that seems fine, I guess. Not having dependency on `git` is nice!

This isn't the case for me. I'm based in Europe, maybe tldr.sh is faster in Europe and github.com is faster in your region (if you're not based in Europe. Accessing both tldr.sh and github.com routes me to north american servers.

**Comment by psibi** [comment]
> I also added an auto-update feature that updates the cache when it is older than 7 days.

Thanks, but I would prefer not have auto-update by default. I'm not opposed to introducing a configuration file where we can control this behavior.

> Is the zip version the slower one?

Yeah. I guess it would also depend on other factors like which commit of git I'm in and so on.

**Comment by gutjuri** [comment]
> > I also added an auto-update feature that updates the cache when it is older than 7 days.
> 
> Thanks, but I would prefer not have auto-update by default. I'm not opposed to introducing a configuration file where we can control this behavior.

I see. I personally don't really like configuration files. I accommodated your wish by making the auto-update optional. Users now have to specify the command line option `--auto-update-interval DAYS` if they wish to enable this functionality. For convenience I also added an explanation on how to set an alias in the command line configuration to have this feature without having to type out the option each time.

File: README.md [hunk 1]
```diff
@@ -57,6 +60,13 @@ Simplified man pages.More information: https://tldr.sh.
    tldr {{git checkout}}
 ```
 
+On the first run, this programme caches all available tldr pages.
```
Comment (psibi) [inline README.md:63]: Probably have a new section named "How it works" ?

Comment (psibi) [inline README.md:63]:
```suggestion
On the first run, this program caches all available tldr pages. 
```

File: src/Tldr/App/Handler.hs [hunk 2]
```diff
@@ -1,39 +1,45 @@
...
   ) where
 
-import Control.Monad (unless)
 import Data.Char (toLower)
 import Data.List (intercalate)
 import Data.Semigroup ((<>))
 import qualified Data.Set as Set
 import Data.Version (showVersion)
+import Data.Time.Clock
+
```
Comment (psibi) [inline src/Tldr/App/Handler.hs:22]: Minor comment: For style consistent there doesn't need to be any space between them.

**Review by psibi — CHANGES_REQUESTED** [review]
This looks awesome! Some minor comments, apart from that LGTM. Thanks for your work!

**Comment by psibi** [comment]
@gutjuri Should I go ahead and finish this MR ?

File: src/Tldr/App/Handler.hs (hunk 2 above)
Comment (gutjuri) [reply src/Tldr/App/Handler.hs]: Not sure what you mean... I committed something, did you mean this?

File: README.md (hunk 1 above)
Comment (gutjuri) [reply README.md]: I think that "How it works" and "Usage" is more or less the same

**Comment by gutjuri** [comment]
> @gutjuri Should I go ahead and finish this MR ?

**Comment by psibi** [comment]
@gutjuri Thanks, I have released a new version with your changes.
